include/ozo/impl/async_request.h

```diff
@@ -326,6 +332,9 @@ struct async_request_op {
     }
 };
 
+template <typename OutHandler, typename Query, typename TimeConstraint, typename Handler>
+async_request_op(Query, TimeConstraint, OutHandler, Handler) -> async_request_op<OutHandler, Query, TimeConstraint, Handler>;
```

Comment:
Note that the constructor arguments here are not in the same order as the template arguments.